### PR TITLE
Fix massive slowdown when editing TileMap

### LINKED ISSUE #61645: Tilemap really slow with few tiles
### Godot version

4.0 alpha 9

### System information

Windows 10, Vulkan, NVIDIA GeForce GTX 1060

### Issue description

The tilemap editor becomes very slow when a certain number of tiles are added (only on alpha 9) someone suggested it could be changes to the HashMap.

Videos to show the difference between the alpha 8 and 9 version:

Alpha 9 ( Slow ) 

https://user-images.githubusercontent.com/16109919/171746421-452c019b-95cf-4757-8dec-7998d736c775.mp4

Alpha 8

https://user-images.githubusercontent.com/16109919/171746493-125f8b91-b4d9-4fcf-bf6b-0a1dab73f6a4.mp4

### Steps to reproduce

Just fill a Tilemap node with a certain number of tiles

### Minimal reproduction project

[BugTilemapAlpha9.zip](https://github.com/godotengine/godot/files/8828085/BugTilemapAlpha9.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #61645

**release_debug** build before this PR:
godot_3s5qieCSEp

**debug** build with this PR:
godot windows tools 64_1AbYSXRoJh

### CODE REVIEW

File: scene/2d/tile_map.h [hunk 1]
```diff
@@ -193,7 +193,7 @@ class TileMap : public Node2D {
 		int y_sort_origin = 0;
 		int z_index = 0;
 		RID canvas_item;
-		HashMap<Vector2i, TileMapCell> tile_map;
+		RBMap<Vector2i, TileMapCell> tile_map;
```
Comment (akien-mga) [inline scene/2d/tile_map.h:196]: I'd add a comment to clarify why RBMap is still used here instead of HashMap.

**Comment by KoBeWi** [comment]
I just noticed that bucket fill still has the slowdown problem. Will try to fix it too.

**Comment by Geometror** [comment]
Did some investigation using Xperf/WPA. As the flame graph shows, a lot of time is wasted in _find/_lookup_pos, even for small/medium tilemaps.
grafik

**Comment by KoBeWi** [comment]
Ok pushed also changes for tile previews.
Went from 2 second lag in release_debug
godot_dMaU6a7wR3
To instant in debug
godot windows tools 64_nxAMwcvLQ3

**Comment by Ansraer** [comment]
Is this superseeded by #61934?

**Comment by KoBeWi** [comment]
If that PR will be merged then yes.

**Comment by akien-mga** [comment]
Superseded by #61934.

**Pull request closed without merging**